src/main/java/com/google/sps/servlets/InterviewerFeedbackServlet.java

```diff
@@ -0,0 +1,133 @@
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+
+// Servlet that gets the feedback from an interviewee and sends it to an interviewer.
+@WebServlet("/interviewer-feedback")
+public class InterviewerFeedbackServlet extends HttpServlet {
+  private ScheduledInterviewDao scheduledInterviewDao;
+  private PersonDao personDao;
+  private Path emailsPath =
+      Paths.get(
+          System.getProperty("user.home") + "/InterviewMe/src/main/resources/templates/email");
+
+  @Override
+  public void init() {
+    init(new DatastoreScheduledInterviewDao(), new DatastorePersonDao());
+  }
+
+  public void init(ScheduledInterviewDao scheduledInterviewDao, PersonDao personDao) {
+    this.scheduledInterviewDao = scheduledInterviewDao;
+    this.personDao = personDao;
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    long scheduledInterviewId = Long.parseLong(request.getParameter("interviewId"));
+    int numberOfQuestions = Integer.parseInt(request.getParameter("questionCount"));
+    HashMap<String, String> answers = new HashMap<String, String>();
+    for (int i = 1; i <= numberOfQuestions; i++) {
+      String template = String.format("{{question_%s}}", i);
+      String param = String.format("question%s", i);
+      answers.put(template, request.getParameter(param));
+    }
+
+    String userEmail = UserServiceFactory.getUserService().getCurrentUser().getEmail();
+    String userId = UserServiceFactory.getUserService().getCurrentUser().getUserId();
+    // Since Users returned from the LocalUserService (in tests) do not have userIds, here we set
+    // the userId equal to a hashcode.
+    if (userId == null) {
+      userId = String.format("%d", userEmail.hashCode());
+    }
+
+    Optional<ScheduledInterview> scheduledInterview = getInterview(scheduledInterviewId);
+    if (!scheduledInterview.isPresent()) {
+      response.sendError(HttpServletResponse.SC_NOT_FOUND);
+      return;
+    }
+
+    answers.put("{{formatted_date}}", scheduledInterview.get().getDateString());
+    if (!isInterviewee(scheduledInterview.get(), userId)) {
+      response.sendError(HttpServletResponse.SC_UNAUTHORIZED);
+      return;
+    }
+
+    if (!getInterviewerEmail(scheduledInterview.get()).isPresent()) {
+      response.sendError(HttpServletResponse.SC_NOT_FOUND);
+      return;
+    }
+
+    try {
+      sendFeedback(getInterviewerEmail(scheduledInterview.get()).get().email(), answers);
+    } catch (Exception e) {
+      e.printStackTrace();
+      response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+      return;
+    }
+    response.sendRedirect("/scheduled-interviews.html");
+    return;
+  }
+
+  private Optional<ScheduledInterview> getInterview(long scheduledInterviewId) {
```

Comment:
This function can probably be deleted with contents inlined with no loss of clarity.

Reply:
Done